test/ollo/grant_types/device_flow_test.exs

```diff
@@ -0,0 +1,274 @@
+defmodule Ollo.GrantTypes.DeviceFlowTest do
+  use ExUnit.Case
+  doctest Ollo.GrantTypes.DeviceFlow
+
+  setup do
+    Application.put_env(:ollo, :allowed_scopes, ["read", "write"])
+  end
+
+  describe ".get_temp_code/1 with a valid client_id" do
+    setup do
+      {:ok, client} = Ollo.register_client(%{name: "Some app"})
+      [client_id: client.client_id]
+    end
+
+    test "returns device and user codes plus their time until expiry", %{client_id: client_id} do
+      {:ok, %{
+        device_code: device_code,
+        user_code: user_code,
+        expires_in: expires_in
```

Comment:
maybe this should be the user's work? in the case of access tokens I made it the user's work.. probably want some consistency there